### PR TITLE
feat: add use of new alert modals for checkout

### PR DESCRIPTION
**Pull request by dalsontws**

### CODE REVIEW

File: src/context/alert.tsx [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+  FunctionComponent,
+  useState,
+  useCallback
+} from "react";
 import {
   AlertModal,
   AlertModalProps
 } from "../components/AlertModal/AlertModal";
 
+export const ERROR_MESSAGE = {
```
Comment (sebastianquek) [inline src/context/alert.tsx:13]: I think all these errors have a period at the end too!

Comment (sebastianquek) [inline src/context/alert.tsx:13]: Same as on the backend, will an enum be appropriate here?

File: src/context/alert.tsx [hunk 2]
```diff
@@ -18,6 +48,64 @@ const defaultAlertProp: AlertModalProps = {
   onExit: () => {}
 };
 
+export const defaultWarningProp: AlertModalProps = {
```
Comment (sebastianquek) [inline src/context/alert.tsx:51]: Maybe can use plural for these?

File: src/hooks/useCart/useCart.tsx [hunk 3]
```diff
@@ -206,7 +207,7 @@ export const useCart = (
             )
           );
         } else {
-          setError(e);
+          setError(new Error(e.message));
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:210]: This will discard the type of error, what's the reason for this change?

File: src/hooks/useCart/useCart.tsx [hunk 4]
```diff
@@ -323,8 +324,15 @@ export const useCart = (
...
+        // backend will throw general duplicate identifier message
+        if (
+          e.message === "Invalid Purchase Request: Duplicate identifier inputs"
+        ) {
+          setError(new Error(ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT));
+        } else {
+          setError(new Error(ERROR_MESSAGE.SERVER_ERROR));
+        }
         setCartState("DEFAULT");
-        setError(new Error("Couldn't checkout, please try again later"));
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:327]: I think we need the error to be thrown after setting the cart state:
https://github.com/rationally-app/mobile-application/pull/129/files

File: src/components/CustomerQuota/CustomerQuotaScreen.tsx [hunk 5]
```diff
@@ -145,11 +144,74 @@ export const CustomerQuotaScreen: FunctionComponent<CustomerQuotaProps> = ({
...
+            description:
+              campaignFeatures?.campaignName === "TT Token"
+                ? ERROR_MESSAGE.INVALID_POD_INPUT
+                : ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT,
+            onOk: () => clearError()
+          });
+          break;
+
+        default:
+          showAlert({
```
Comment (sebastianquek) [inline src/components/CustomerQuota/CustomerQuotaScreen.tsx:207]: Do you think a sentry error here will be useful so we know there's an unhandled error? Or should we just throw and let the error boundary take care of it?

File: src/hooks/useCart/useCart.tsx [hunk 6]
```diff
@@ -249,7 +250,7 @@ export const useCart = (
   const updateCart: CartHook["updateCart"] = useCallback(
     (category, quantity, identifierInputs) => {
       if (quantity < 0) {
-        setError(new Error("Invalid quantity"));
+        setError(new Error(ERROR_MESSAGE.INVALID_QUANTITY));
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:253]: This works well for now. Am wondering if it's useful to throw a special `CartError` instead? Then we can scope the error messages to be under `CartError` i.e. `new CartError(INVALID_QUANTITY)`. Perhaps the alert props can be inside the error too.

**Review by sebastianquek — CHANGES_REQUESTED** [review]

File: src/hooks/useCart/useCart.tsx (hunk 4 above)
Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx:327]: this is actually just setting the error, not throwing it. would this be fine?

File: src/context/alert.tsx (hunk 1 above)
Comment (dalsontws) [reply src/context/alert.tsx:13]: oh! i was referencing the previous error messages that didnt have a period behind! should we add that?

Comment (dalsontws) [reply src/context/alert.tsx]: good idea! changed!

File: src/components/CustomerQuota/CustomerQuotaScreen.tsx (hunk 5 above)
Comment (dalsontws) [reply src/components/CustomerQuota/CustomerQuotaScreen.tsx]: alright! will be throwing to error boundary!

File: src/context/alert.tsx (hunk 2 above)
Comment (dalsontws) [reply src/context/alert.tsx]: got it, added!

File: src/hooks/useCart/useCart.tsx (hunk 6 above)
Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx:253]: good idea for this! will add this as a task for the bigger PR, thanks!

File: src/context/alert.tsx (hunk 1 above)
Comment (dalsontws) [reply src/context/alert.tsx]: added periods after asking Eida!

File: src/hooks/useCart/useCart.tsx (hunk 3 above)
Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx]: reverting to use `setError(e)`, will be thrown to error boundary!

File: src/hooks/useCart/useCart.tsx (hunk 4 above)
Comment (sebastianquek) [reply src/hooks/useCart/useCart.tsx:327]: Oh like it needs to be set after the cart state haha

Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx:327]: okay got it and changed!

**Review by sebastianquek — APPROVED** [review]
LGTM!
